TLM/TLM/Util/Extensions/NetSegmentExtensions.cs

```diff
@@ -128,6 +129,34 @@ public static class NetSegmentExtensions {
             return false;
         }
 
+        /// <summary>
+        /// Count the number of lanes matching specified criteria at a segment end.
+        /// </summary>
+        /// <param name="segment">The segment to inspect.</param>
+        /// <param name="nodeId">The id of the node to inspect.</param>
+        /// <param name="laneTypeFilter">Optionally filter to specified lane types.</param>
+        /// <param name="vehicleTypeFilter">Optionally filter to specified vehicle types.</param>
+        /// <param name="incoming">
+        /// If <c>true</c>, count lanes entering the segment via the node.
+        /// if <c>false</c>, count lanes leaving the segment via the node.
+        /// </param>
+        /// <returns>Returns number of lanes matching specified criteria.</returns>
```

Comment:
this has not been used in critical part of the code. but if it is used then it needs improvements.
Either optimize or put a comment about performance..

Reply:
It's literally the same code that was used in all the places that now direct to this extension.

Additionally, unlike most of the original call sites, it ensures no lane sorting takes place. And now that the code block is centralised any perf improvements will benefit all call sites.

In any case, I can see some obvious improvements here - mainly by ditching the call to `GetSortedLanes` and using more specific code, so I'll send a commit with those changes shortly.

Reply:
Updated to remove `GetSortedLanes` overhead from `CountLanes`.